### PR TITLE
docs: Updating Azure CNI chaining as Legacy approach

### PR DESCRIPTION
**Pull request by vipul-21**
Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

Related: https://github.com/cilium/cilium/issues/27289

### CODE REVIEW

**Review by tamilmani1989 — APPROVED** [review]

**Comment by vipul-21** [comment]
Depends on this PR changes: https://github.com/cilium/cilium/pull/28286

File: Documentation/installation/cni-chaining-azure-cni.rst
Comment (tommyp1ckles) [inline Documentation/installation/cni-chaining-azure-cni.rst]: nit: seems like this could be two sentences ... i.e. in chaining configuration.  This is the legacy way...

**Review by tommyp1ckles — APPROVED** [review]

File: Documentation/installation/cni-chaining-azure-cni.rst
Comment (qmonnet) [inline Documentation/installation/cni-chaining-azure-cni.rst]: Nit: please wrap the line.

```suggestion
   The best way to run cilium on AKS is through :ref:`aks_install` or
   `Azure CNI Powered by Cilium <https://aka.ms/aks/cilium-dataplane>`__.
```

File: Documentation/installation/cni-chaining-azure-cni.rst
Comment (qmonnet) [inline Documentation/installation/cni-chaining-azure-cni.rst]:
```suggestion
   cilium as Azure IPAM is legacy, for more information see :ref:`ipam_azure`.
```

**Review by qmonnet — CHANGES_REQUESTED** [review]

**Comment by nbusseneau** [comment]
I am switching PRs related to #27289 to draft so that I have a chance to review them cohesively since they're all out at the same time and are all supposed to work together.

**Comment by dylandreimerink** [comment]
@vipul-21 CI seems to have had a bad day last time checking the PR, I suggest doing a rebase against main to re-trigger the CI so that the PR might proceed.

**Review by qmonnet — APPROVED** [review]

**Comment by vipul-21** [comment]
Can we merge this PR ? @tommyp1ckles @qmonnet

**Comment by qmonnet** [comment]
> Can we merge this PR ?

The PR is still waiting for a review from Nicolas, who wanted to take a look, but I understand he's out for some time. The change is small enough that I think we're good to go without his ACK, though.

One question before we merge, I see the following:

> This PR depends on the doc PR: https://github.com/cilium/cilium/pull/27199

What's the status on that?

**Comment by squeed** [comment]
I see that #27199, listed as a dependency, has been closed instead of merged. Does this PR still rely on that? If I can merge this.

**Comment by vipul-21** [comment]
@squeed Removed the dependency as this PR can be merged now. Thanks !

**Pull request merged**
